File: RemoveElement.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */

var removeElement = function(nums, val) {
    let j=nums.length-1,i=0;
    while(i<=j)
    {
        if(nums[i]==val)
        {
            nums[i]=nums[j];
            nums[j]=val;
            j--;
        }
        else
        {
            i++
        }
    }
    return i;
};

File: RemoveElement.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int removeElement(vector<int>& nums, int val) 
    {
        int i=0;
        auto itr=nums.begin();
        while(i<nums.size())
        {
            
            if(nums[i]==val)
            {
                nums.erase(itr+i);
            }
            else
            {
                i++;
            }
        }
        return i;
    }
    
};

int main()
{
    
    return 0;
}
